include/uv.h

```diff
@@ -243,6 +243,11 @@ typedef enum {
 UV_EXTERN unsigned int uv_version(void);
 UV_EXTERN const char* uv_version_string(void);
 
+typedef void* (*uv_malloc_cb)(size_t size);
+typedef void (*uv_free_cb)(void* ptr);
+
+UV_EXTERN void uv_replace_allocator(uv_malloc_cb malloc_cb, uv_free_cb free_cb);
```

Comment:
Great, thanks! Don't do the single function thing for now, it was just a
quick suggestion I'd like more input on.
On Feb 3, 2015 6:44 PM, "Brett Vickers" <notifications@github.com> wrote:

> In include/uv.h
> <https://github.com/libuv/libuv/pull/178#discussion_r24022766>:
>
> > @@ -243,6 +243,11 @@ typedef enum {
> >  UV_EXTERN unsigned int uv_version(void);
> >  UV_EXTERN const char* uv_version_string(void);
> >
> > +typedef void* (*uv_malloc_cb)(size_t size);
> > +typedef void (*uv_free_cb)(void* ptr);
> > +
> > +UV_EXTERN void uv_replace_allocator(uv_malloc_cb malloc_cb, uv_free_cb free_cb);
>
> As someone who is often "that guy", I totally get it. :-) I'm happy to use
> the func prefix. Will make that change.
>
> Re: a single allocator function for free and malloc, I'm happy to do that
> too if you don't think it'll be too wonky. It's the kind of thing that
> would probably require a non-trivial comment explaining it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/libuv/libuv/pull/178/files#r24022766>.
>